### ISSUE TITLE
Allow users to be banned

### ISSUE THREAD

**Issue by rjollos** [trac-hacks/trac-github#123]
Sometimes we get users generating spam with their GitHub accounts. I guess there's probably a way to report these users, but it would be nice if we could just ban users by username. So far we haven't had a problem since these users go away after making a few comments and we are running TracSpamFilter so some stuff gets trapped. It would probably be simple enough though to have a list of users on a wiki page, and to reject login by users matching one of the usernames.

**Comment by aaugustin** [comment]
Does this need to be coupled with trac-github? It seems to me that it could live as a standalone plugin, which may be more widely useful?

**Comment by rjollos** [comment]
It won't be very useful as a standalone plugin. It's only useful in the case that you have a `LoginModule` that doesn't have any control over the username or password store. It will also be awkward to implement in a plugin and is only going to be a few lines of code. However, if you don't want it in trac-github I will find another way.

**Comment by aaugustin** [comment]
I see, you want to blacklist the github username rather than the trac username.

You can go ahead with this idea.

**Comment by rjollos** [comment]
Right, I was thinking to just compare the [username](https://github.com/trac-hacks/trac-github/blob/master/tracext/github.py#L141) and reject the login if the name is in the list of blacklisted users.
